integration-cli/docker_cli_network_unix_test.go

```diff
@@ -351,6 +353,8 @@ func (s *DockerNetworkSuite) TestDockerNetworkRmPredefined(c *check.C) {
 }
 
 func (s *DockerNetworkSuite) TestDockerNetworkLsFilter(c *check.C) {
+	// E2E: Test assumes clean network environment.
+	testRequires(c, NotE2E)
```

Comment:
This test is unfortunately not written very well because it actually tests many cases in one.

Instead of skipping the entire function could we put an `if CleanEnvironment()` around the cases that don't work?

`CleanEnvironment()` can be similar to `NotE2E` but would use a different env variable name.

Reply:
With the other changes you suggested, I've removed the `E2E` requirement. For this test, I added an `OnlyDefaultNetworks` requirement as that's more precise.